### ISSUE TITLE
Option for square icons

### ISSUE THREAD

**Issue by mlaggner** [JFormDesigner/FlatLaf#118]
I saw that you implemented making icon only JButton not being square with #110, but I exactly need this behavior for a button (just containing 1 icon of font awesome).  

I've gone through the code but did not see any hint how I could make a single icon button being rendered square without subclassing the FlatButtonUI - did I miss something?

Thanks again

**Comment by DevCharly** [comment]
Yep, fix for #110 was probably "not optimal"...

What library do you use? [jIconFont](http://jiconfont.github.io/swing), [ikonli](https://github.com/kordamp/ikonli) or another one?

**Comment by mlaggner** [comment]
I use the SVGs directly via SvgSalamander - so my JButton is basically just a button with no text and exactly 1 Icon

**Comment by DevCharly** [comment]
@mlaggner I've added a client property in commit 7b11e291228b6f582d486fe4e75353f584826f1e that allows making any button square:
~~~java
myButton.putClientProperty( "JButton.squareSize", true );
or
myButton.putClientProperty( FlatClientProperties.SQUARE_SIZE, true );
~~~

Thought about having some mechanism that makes icon-only buttons automatically square, but did not find a practical solution. In most cases the icons are already square. In some cases they are rectangular and require also rectangular buttons. In cases where small icons are non-square and should result in square buttons its better to explicitly set the client property.

BTW there is another easy way to get square buttons from non-square icons:
~~~java
class SquareIcon implements javax.swing.Icon {
    Icon icon;

    SquareIcon( Icon icon ) {
        this.icon = icon;
    }

    int getIconWidth() {
        return Math.max( icon.getIconWidth(), icon.getIconHeight() );
    }

    int getIconHeight() {
         return Math.max( icon.getIconWidth(), icon.getIconHeight() );
    }

    void paintIcon(Component c, Graphics g, int x, int y) {
        x += (getIconWidth() - icon.getIconWidth()) / 2;
        y += (getIconHeight() - icon.getIconHeight()) / 2;
        icon.paintIcon( c, g, x, y );
    }
}
~~~

**Comment by DevCharly** [comment]
The nice thing about the new "make square" client property is that it is now easy to create circle buttons:

![image](https://user-images.githubusercontent.com/5604048/86008451-12c3ec00-ba19-11ea-915c-9bfa57d73459.png)

~~~java
myButton.putClientProperty( "JButton.squareSize", true );
myButton.putClientProperty( "JButton.buttonType", "roundRect" );
or
myButton.putClientProperty( FlatClientProperties.SQUARE_SIZE, true );
myButton.putClientProperty( FlatClientProperties.BUTTON_TYPE, FlatClientProperties.BUTTON_TYPE_ROUND_RECT );
~~~

**Comment by DevCharly** [comment]
this is now released in [FlatLaf 0.37](https://github.com/JFormDesigner/FlatLaf/releases/tag/0.37)

**Comment by mlaggner** [comment]
awesome! thanks
